### ISSUE TITLE
A way of making SetPreventTransmit override the transmission status of the entity's children

### ISSUE THREAD

**Issue by WilliamVenner** [Facepunch/garrysmod-requests#1528]
Currently, it is extremely difficult to accurately control the transmission status of a `Player` entity.

Preventing transmission on a player doesn't do anything at first, until you also prevent its children from transmitting.

This can be extremely hacky to deal with - just to prevent a player from transmitting you have to track entity creations, weapon switching, respawning, etc. and call `SetPreventTransmit` an unacceptable amount of times throughout the server's lifetime to control this.

One example of some extremely finnicky behaviour here is the god awful `physgun_beam` entity. During the `OnEntityCreated` hook, `physgun_beam` obviously has no parent. So you have no idea who the `physgun_beam` actually belongs to. This means, that if you do successfully get a player to stop transmitting to others, it's almost impossible to enforce this transmission status when they switch to their physgun without some hacky timers checking the player's children.

Additionally, there's no hook for when an entity gets a new child, or when an entity gets a new parent.

**TL;DR:** It would be really useful if there was some kind of way of making `SetPreventTransmit` override/ignore the transmission status of the entity's children, so that you can much more efficiently prevent transmission.

**Comment by WilliamVenner** [comment]
I've just noticed that calling `SetTransmitWithParent(true)` on `physgun_beam` has no effect whatsoever. If I call that, I still have to call `SetPreventTransmit(true, ply)` for it to stop transmitting me.

**Comment by BillysDOG** [comment]
ok BOOMER

**Comment by DingoDown** [comment]
This would be really useful!
